src/test/java/core/basesyntax/service/InputDataServiceImplTest.java

```diff
@@ -0,0 +1,89 @@
+package core.basesyntax.service;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+import java.util.List;
+import org.junit.Test;
+
+public class InputDataServiceImplTest {
+    @Test
+    public void stringToFruitTransactionConverter_validInputTextFormat_ok() {
+        InputDataService inputDataService = new InputDataServiceImpl();
+        String validLineOne = "b,pineapple,10";
+        String validLineTwo = "r,banana,100";
+        String validLineThree = "p,orange,50";
+        List<String> validInputTextList = List.of(validLineOne, validLineTwo, validLineThree);
+        assertEquals(3,
+                inputDataService.stringToFruitTransactionConverter(validInputTextList).size());
+    }
+
+    @Test
+    public void stringToFruitTransactionConverter_invalidDataFormatExceptionMessage_Ok() {
+        InputDataService inputDataService = new InputDataServiceImpl();
+        String invalidLineOne = "pineapple,10";
+        String invalidLineTwo = "r,100";
+        String invalidLineThree = "p,orange";
+        String emptyLine = "";
+
+        try {
+            inputDataService.stringToFruitTransactionConverter(List.of(invalidLineOne));
+            fail("Exception not thrown");
```

Comment:
Here and below - add details to the fail message. We should understand what was wrong.